### PR TITLE
Patch nn.TransformerEncoder into HF

### PR DESCRIPTION
**Pull request by erichan1**
Steps to use the benchmark:

1. Install PyTorch nightly to get access to the fast path of nn.TransformerEncoder.
`pip3 install --pre torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/nightly/cu113`

2. Install this version of HF transformers by cloning this branch and running this cmd from the transformers dir
`pip install -e .`

3. Navigate to benchmark_bettertransformer.py and run `python3 benchmark_bettertransformer.py`

### CODE REVIEW

File: src/transformers/models/bert/modeling_bert.py [hunk 1]
```diff
@@ -526,13 +595,95 @@ def feed_forward_chunk(self, attention_output):
...
+            raise Exception("Does not support")
+
+        all_hidden_states = () if output_hidden_states else None
+        all_self_attentions = () if output_attentions else None
+
+        if attention_mask is not None:
+            attention_mask = ~attention_mask.bool()
+            attention_mask = torch.reshape(attention_mask, (attention_mask.shape[0], attention_mask.shape[-1]))
+            # remove padding from hidden states
+            # hidden_states = torch.nested_tensor(hidden_states)
```
Comment (erichan1) [inline src/transformers/models/bert/modeling_bert.py]: create a NestedTensor here if a mask is passed? And we'd remove padding after if the mask is passed - just haven't put it into code yet.
